apps/omg_api/lib/root_chain_coordinator/core.ex

```diff
@@ -119,13 +137,12 @@ defmodule OMG.API.RootChainCoordinator.Core do
     end
   end
 
-  defp all_services_checked_in?(state) do
-    registered =
-      state.services
-      |> Map.keys()
-      |> MapSet.new()
+  defp get_synced_height_by_mode(state, :sync_with_root_chain) do
+    {:sync, state.root_chain_height}
+  end
 
-    state.allowed_services == registered
+  defp all_services_checked_in?(state) do
+    state.configs_services |> Map.keys() == state.services |> Map.keys()
```

Comment:
you assume that heys will be in this same order

Reply:
Good catch!